smithy-aws-protocol-tests/model/restXml/services/s3.smithy

```diff
@@ -272,6 +274,47 @@ operation ListObjectsV2 {
 }
 
 
+@httpRequestTests([
+    {
+        id: "S3EscapeObjectKeyInUriLabel",
+        documentation: """
+            S3 clients should escape special characters in Object Keys
+            when the Object Key is used as a URI label binding.
+        """,
+        protocol: restXml,
+        method: "DELETE",
+        uri: "/my%20key.txt",
+        host: "s3.us-west-2.amazonaws.com",
+        resolvedHost: "mybucket.s3.us-west-2.amazonaws.com",
+        body: "",
+        queryParams: [
+            "tagging"
+        ],
+        params: {
+            Bucket: "mybucket",
+            Key: "my key.txt"
```

Comment:
Can we add another test with a path in the key? Something like `foo/bar/my key.txt`. I'd expect this to be encoded as `foo/bar/my%20key.txt` and not `foo%2Fbar%2Fmy%20key.txt`, is that the case?

Reply:
Your expectation is correct, I agree that's a good test to add, thanks for calling it out. I pushed an update including that test.